### PR TITLE
V8: Hide "umbContent" app if the content has no properties

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/4789

### Description

If a content type does not have any properties, the "umbContent" app should not be shown. This PR fixes it:

content-without-properties

#### Testing this PR

Given a content type with no properties, ensure that:

1. The "umbContent" app is not displayed when editing content of this type.
2. Editing content of this type works both with and without listview enabled. 
   - Without listview enabled, no apps should be shown as there's only one app available ("umbInfo").
3. You can't create a content template for content of this type from settings (it makes no sense as there are no properties).
4. Creating a content template from existing content of this type is still possible, but the resulting content template is uneditable (it has no properties).

For all test cases above, also ensure that things work normally for content types *with* properties.

### CODE REVIEW

**Comment by emmaburstow** [comment]
Hiya @kjac!

Thanks for this. We'll let you know if we need anything.

Em

**Comment by kjac** [comment]
Merge conflict resolved.

**Comment by poornimanayar** [comment]
Hey Kenn,

I am gonna test this shortly :-)

Poornima

**Comment by poornimanayar** [comment]
Great work @kjac! Have tested all the scenarios mentioned and it works very well! Great idea implemented and very helpful testing tips as well!

File: src/Umbraco.Web.UI.Client/src/views/components/content/umb-variant-content.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     <div class="umb-split-view__content" ng-show="!vm.editor.loading">
 
-        <ng-form name="contentHeaderForm">
+        <ng-form name="contentHeaderForm" ng-if="vm.editor.content.apps.length > 0">
```
Comment (poornimanayar) [inline src/Umbraco.Web.UI.Client/src/views/components/content/umb-variant-content.html]: Wondering if the ng-if can be changed to ng-show?

Comment (kjac) [reply src/Umbraco.Web.UI.Client/src/views/components/content/umb-variant-content.html]: Nope... for some reason stuff breaks if the `ng-form` is there but just hidden. I did try 😄

Comment (poornimanayar) [reply src/Umbraco.Web.UI.Client/src/views/components/content/umb-variant-content.html]: I completely thought so, but just asking :)

**Comment by nul800sebastiaan** [comment]
One additional thing to check for this one @kjac @poornimanayar - if there is a custom content app available for this doctype (really, there shouldn't be but... ) then does the custom content app still show? I guess it COULD still be useful if the custom content app does something based on the title of the page or the scheduled publish date or something.

Other than that this looks FANASTIC! 👍

**Comment by kjac** [comment]
@nul800sebastiaan I would certainly expect the custom content app to show regardless of the presence of "umbContent" app. It would be part of the collection of content apps generated serverside and thus send to the client as just another content app to show.

The custom content app in question must then be smart enough to handle the lack of properties on the content. 

I'm certain one could even think up a scenario where it would make sense to have a custom content app specifically for content with no properties.

**Comment by poornimanayar** [comment]
Hi @nul800sebastiaan and @kjac ,

Tested the scenario around Content Apps and its looking good as well :-)

Poornima

**Comment by nul800sebastiaan** [comment]
All good, thanks Kenn! 🌟

**Pull request merged**